geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/RegionCreateFunction.java

```diff
@@ -149,29 +147,90 @@ private CliFunctionResult handleException(final String memberNameOrId, final Str
     return new CliFunctionResult(memberNameOrId, CliFunctionResult.StatusState.ERROR);
   }
 
-  private <K, V> Region<?, ?> createRegion(Cache cache, RegionFunctionArgs regionCreateArgs) {
-    Region<K, V> createdRegion = null;
-
-    // create the region factory using the arguments
-    RegionAttributes<K, V> regionAttributes = regionCreateArgs.getRegionAttributes();
-    boolean isPartitioned = regionAttributes.getPartitionAttributes() != null;
-    RegionFactory<K, V> factory = cache.createRegionFactory(regionAttributes);
+  private <K, V> Region<?, ?> createRegion(Cache cache, RegionConfig config, String regionPath)
```

Comment:
Given this function is already at 250 lines and there are lots of small independent sections of code, maybe you can bring this out into a series of smaller private methods. It would make this function easier to read

Reply:
Sure, that's a refactor we could do.